server/src/main/java/org/elasticsearch/index/mapper/RootObjectMapper.java

```diff
@@ -168,6 +168,9 @@ protected boolean processField(RootObjectMapper.Builder builder, String fieldNam
                       }
                   ]
                 */
+                if (!(fieldNode instanceof List)) {
```

Comment:
small style nit: can you use `((fieldNode instanceof List) == false)` instead? We prefer that kind of more verbose state to the negate operator for better readability throughout the codebase.

Reply:
@cbuescher ok I will fix for you suggest